configs/MERCURYH7/config.h

```diff
@@ -0,0 +1,118 @@
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define FC_TARGET_MCU     STM32H743
+
+#define BOARD_NAME                      MERCURYH7
+#define MANUFACTURER_ID                 ANYL
+
+#define USE_ACC
+#define USE_ACC_SPI_ICM42688P
+#define USE_GYRO
+#define USE_GYRO_SPI_ICM42688P
+#define USE_BARO
+#define USE_BARO_DPS310
+#define USE_FLASH
+#define USE_FLASH_M25P16
+
+#define BEEPER_PIN                      PE9
+#define MOTOR1_PIN                      PC6
+#define MOTOR2_PIN                      PC7
+#define MOTOR3_PIN                      PC8
+#define MOTOR4_PIN                      PC9
+#define MOTOR5_PIN                      PE5
+#define MOTOR6_PIN                      PE6
+#define MOTOR7_PIN                      PE13
+#define MOTOR8_PIN                      PE14
+#define UART1_TX_PIN                    PB6
+#define UART2_TX_PIN                    PA2
+#define UART3_TX_PIN                    PD8
+#define UART4_TX_PIN                    PC10
+#define UART7_TX_PIN                    PB4
+#define UART8_TX_PIN                    PE1
+#define UART1_RX_PIN                    PB7
+#define UART2_RX_PIN                    PA3
+#define UART3_RX_PIN                    PD9
+#define UART4_RX_PIN                    PC11
+#define UART7_RX_PIN                    PB3
+#define UART8_RX_PIN                    PE0
+#define I2C1_SCL_PIN                    PB8
+#define I2C1_SDA_PIN                    PB9
+#define I2C2_SCL_PIN                    PB10
+#define I2C2_SDA_PIN                    PB11
+#define SPI1_SCK_PIN                    PA5
+#define SPI2_SCK_PIN                    PA9
+#define SPI1_SDI_PIN                    PA6
+#define SPI2_SDI_PIN                    PB14
+#define SPI1_SDO_PIN                    PA7
+#define SPI2_SDO_PIN                    PB15
+#define ADC_VBAT_PIN                    PA4
+#define ADC_CURR_PIN                    PA0
+#define FLASH_CS_PIN                    PE11
+#define GYRO_1_EXTI_PIN                 PB12
+#define GYRO_1_CS_PIN                   PC4
+#define LED0_PIN                        PC13
+
+#define TIMER_PIN_MAPPING               TIMER_PIN_MAP( 1, MOTOR1_PIN , 1,  -1) \
+                                        TIMER_PIN_MAP( 2, MOTOR2_PIN , 1,  -1) \
+                                        TIMER_PIN_MAP( 3, MOTOR3_PIN , 1,  -1) \
+                                        TIMER_PIN_MAP( 4, MOTOR4_PIN , 1,  -1) \
+                                        TIMER_PIN_MAP( 5, MOTOR5_PIN , 1,  -1) \
+                                        TIMER_PIN_MAP( 6, MOTOR6_PIN , 1,  -1) \
+                                        TIMER_PIN_MAP( 7, MOTOR7_PIN , 1,  -1) \
+                                        TIMER_PIN_MAP( 8, MOTOR8_PIN , 1,  -1) \
+                                        TIMER_PIN_MAP( 0, BEEPER_PIN , 2,  -1)
```

Comment:
index is wrong. Also no dma even for motors?

Reply:
How would you correct this?

Reply:
Here is all the info that is relevant from a hardware perspective (Other than the pins, which are defined in this PR)

Motors 1-4: Tim3, CH1-4
Motors 5-6: Tim15, CH1-2
Motors 7-8: TIM1, CH3-4

Ideally, Motors 1-4 are on the same DMA channel as burst DMA, but I understand Burst DMA isn't implemented in an efficient way in BF, so I'm happy setting this up in whatever config you'd like. I don't understand the TIMER_PIN_MAPPING section, or how the DMA setup works, since it tends to ignore what I set for motor DMA.

Reply:
Can help with the TIMER_PIN_MAPPING if i know the dma and timer output from cli

Reply:
`dma show`:
```
Currently active DMA:
--------------------
DMA1 Stream 0: DSHOT_BITBANG 3
DMA1 Stream 1: FREE
DMA1 Stream 2: FREE
DMA1 Stream 3: FREE
DMA1 Stream 4: FREE
DMA1 Stream 5: FREE
DMA1 Stream 6: ADC 1
DMA1 Stream 7: FREE
DMA2 Stream 0: SPI_SDO 1
DMA2 Stream 1: SPI_SDI 1
DMA2 Stream 2: SPI_SDO 2
DMA2 Stream 3: SPI_SDI 2
DMA2 Stream 4: FREE
DMA2 Stream 5: FREE
DMA2 Stream 6: FREE
DMA2 Stream 7: FREE
```

`dump`
```
# dump

# version
# Betaflight / STM32H743 (SH74) 4.5.0 Jan 17 2024 / 16:05:47 (926ef3447) MSP API: 1.46
# config rev: 8c5cd95

# start the command batch
batch start

board_name MERCURYH7
manufacturer_id ANYL

# name: -

# resources
resource BEEPER 1 E09
resource MOTOR 1 C06
resource MOTOR 2 C07
resource MOTOR 3 C08
resource MOTOR 4 C09
resource MOTOR 5 E05
resource MOTOR 6 E06
resource MOTOR 7 E13
resource MOTOR 8 E14
resource SERVO 1 NONE
resource SERVO 2 NONE
resource SERVO 3 NONE
resource SERVO 4 NONE
resource SERVO 5 NONE
resource SERVO 6 NONE
resource SERVO 7 NONE
resource SERVO 8 NONE
resource PPM 1 NONE
resource PWM 1 NONE
resource PWM 2 NONE
resource PWM 3 NONE
resource PWM 4 NONE
resource PWM 5 NONE
resource PWM 6 NONE
resource PWM 7 NONE
resource PWM 8 NONE
resource SONAR_TRIGGER 1 NONE
resource SONAR_ECHO 1 NONE
resource LED_STRIP 1 NONE
resource SERIAL_TX 1 B06
resource SERIAL_TX 2 A02
resource SERIAL_TX 3 D08
resource SERIAL_TX 4 C10
resource SERIAL_TX 5 NONE
resource SERIAL_TX 6 NONE
resource SERIAL_TX 7 B04
resource SERIAL_TX 8 E01
resource SERIAL_TX 9 NONE
resource SERIAL_TX 10 NONE
resource SERIAL_TX 11 NONE
resource SERIAL_RX 1 B07
resource SERIAL_RX 2 A03
resource SERIAL_RX 3 D09
resource SERIAL_RX 4 C11
resource SERIAL_RX 5 NONE
resource SERIAL_RX 6 NONE
resource SERIAL_RX 7 B03
resource SERIAL_RX 8 E00
resource SERIAL_RX 9 NONE
resource SERIAL_RX 10 NONE
resource SERIAL_RX 11 NONE
resource SOFTSERIAL_TX 1 NONE
resource SOFTSERIAL_TX 2 NONE
resource SOFTSERIAL_RX 1 NONE
resource SOFTSERIAL_RX 2 NONE
resource I2C_SCL 1 B08
resource I2C_SCL 2 B10
resource I2C_SCL 3 NONE
resource I2C_SCL 4 NONE
resource I2C_SDA 1 B09
resource I2C_SDA 2 B11
resource I2C_SDA 3 NONE
resource I2C_SDA 4 NONE
resource LED 1 C13
resource LED 2 NONE
resource LED 3 NONE
resource RX_BIND 1 NONE
resource RX_BIND_PLUG 1 NONE
resource TRANSPONDER 1 NONE
resource SPI_SCK 1 A05
resource SPI_SCK 2 A09
resource SPI_SCK 3 NONE
resource SPI_SCK 4 NONE
resource SPI_SCK 5 NONE
resource SPI_SCK 6 NONE
resource SPI_SDI 1 A06
resource SPI_SDI 2 B14
resource SPI_SDI 3 NONE
resource SPI_SDI 4 NONE
resource SPI_SDI 5 NONE
resource SPI_SDI 6 NONE
resource SPI_SDO 1 A07
resource SPI_SDO 2 B15
resource SPI_SDO 3 NONE
resource SPI_SDO 4 NONE
resource SPI_SDO 5 NONE
resource SPI_SDO 6 NONE
resource ESCSERIAL 1 NONE
resource ADC_BATT 1 A04
resource ADC_RSSI 1 NONE
resource ADC_CURR 1 A00
resource ADC_EXT 1 NONE
resource BARO_CS 1 NONE
resource BARO_EOC 1 NONE
resource BARO_XCLR 1 NONE
resource PINIO 1 NONE
resource PINIO 2 NONE
resource PINIO 3 NONE
resource PINIO 4 NONE
resource USB_MSC_PIN 1 NONE
resource FLASH_CS 1 E11
resource GYRO_EXTI 1 B12
resource GYRO_EXTI 2 NONE
resource GYRO_CS 1 C04
resource GYRO_CS 2 NONE
resource USB_DETECT 1 NONE
resource VTX_POWER 1 NONE
resource VTX_CS 1 NONE
resource VTX_DATA 1 NONE
resource VTX_CLK 1 NONE
resource PULLUP 1 NONE
resource PULLUP 2 NONE
resource PULLUP 3 NONE
resource PULLUP 4 NONE
resource PULLDOWN 1 NONE
resource PULLDOWN 2 NONE
resource PULLDOWN 3 NONE
resource PULLDOWN 4 NONE

# timer
timer E09 AF192
# pin E09: TIM0 CH63 (AF192)
timer C06 AF2
# pin C06: TIM3 CH1 (AF2)
timer C07 AF2
# pin C07: TIM3 CH2 (AF2)
timer C08 AF2
# pin C08: TIM3 CH3 (AF2)
timer C09 AF2
# pin C09: TIM3 CH4 (AF2)
timer E05 AF4
# pin E05: TIM15 CH1 (AF4)
timer E06 AF4
# pin E06: TIM15 CH2 (AF4)
timer E13 AF1
# pin E13: TIM1 CH3 (AF1)
timer E14 AF1
# pin E14: TIM1 CH4 (AF1)

# dma
dma SPI_SDO 1 8
# SPI_SDO 1: DMA2 Stream 0 Request 38
dma SPI_SDO 2 10
# SPI_SDO 2: DMA2 Stream 2 Request 40
dma SPI_SDO 3 NONE
dma SPI_SDO 4 NONE
dma SPI_SDO 5 NONE
dma SPI_SDO 6 NONE
dma SPI_SDI 1 9
# SPI_SDI 1: DMA2 Stream 1 Request 37
dma SPI_SDI 2 11
# SPI_SDI 2: DMA2 Stream 3 Request 39
dma SPI_SDI 3 NONE
dma SPI_SDI 4 NONE
dma SPI_SDI 5 NONE
dma SPI_SDI 6 NONE
dma SPI_TX 1 8
# SPI_TX 1: DMA2 Stream 0 Request 38
dma SPI_TX 2 10
# SPI_TX 2: DMA2 Stream 2 Request 40
dma SPI_TX 3 NONE
dma SPI_TX 4 NONE
dma SPI_TX 5 NONE
dma SPI_TX 6 NONE
dma SPI_RX 1 9
# SPI_RX 1: DMA2 Stream 1 Request 37
dma SPI_RX 2 11
# SPI_RX 2: DMA2 Stream 3 Request 39
dma SPI_RX 3 NONE
dma SPI_RX 4 NONE
dma SPI_RX 5 NONE
dma SPI_RX 6 NONE
dma ADC 1 6
# ADC 1: DMA1 Stream 6 Request 9
dma ADC 2 NONE
dma ADC 3 NONE
dma UART_TX 1 NONE
dma UART_TX 2 NONE
dma UART_TX 3 NONE
dma UART_TX 4 NONE
dma UART_TX 5 NONE
dma UART_TX 6 NONE
dma UART_TX 7 NONE
dma UART_TX 8 NONE
dma UART_TX 9 NONE
dma UART_TX 10 NONE
dma UART_TX 11 NONE
dma UART_RX 1 NONE
dma UART_RX 2 NONE
dma UART_RX 3 NONE
dma UART_RX 4 NONE
dma UART_RX 5 NONE
dma UART_RX 6 NONE
dma UART_RX 7 NONE
dma UART_RX 8 NONE
dma UART_RX 9 NONE
dma UART_RX 10 NONE
dma UART_RX 11 NONE
dma TIMUP 1 NONE
dma TIMUP 2 NONE
dma TIMUP 3 NONE
dma TIMUP 4 NONE
dma TIMUP 5 NONE
dma TIMUP 6 NONE
dma TIMUP 7 NONE
dma TIMUP 8 NONE
dma TIMUP 12 NONE
dma TIMUP 13 NONE
dma TIMUP 14 NONE
dma TIMUP 15 NONE
dma TIMUP 16 NONE
dma TIMUP 17 NONE
dma TIMUP 0 NONE
dma TIMUP 0 NONE
dma TIMUP 0 NONE
dma pin E09 NONE
dma pin C06 NONE
dma pin C07 NONE
dma pin C08 NONE
dma pin C09 NONE
dma pin E05 NONE
dma pin E06 NONE
dma pin E13 NONE
dma pin E14 NONE

# feature
feature -RX_PPM
feature -INFLIGHT_ACC_CAL
feature -RX_SERIAL
feature -MOTOR_STOP
feature -SERVO_TILT
feature -SOFTSERIAL
feature -GPS
feature -RANGEFINDER
feature -TELEMETRY
feature -3D
feature -RX_PARALLEL_PWM
feature -RX_MSP
feature -RSSI_ADC
feature -LED_STRIP
feature -DISPLAY
feature -OSD
feature -CHANNEL_FORWARDING
feature -TRANSPONDER
feature -AIRMODE
feature -RX_SPI
feature -ESC_SENSOR
feature -ANTI_GRAVITY
feature RX_SERIAL
feature SERVO_TILT
feature RANGEFINDER
feature TELEMETRY
feature LED_STRIP
feature DISPLAY
feature OSD
feature CHANNEL_FORWARDING
feature TRANSPONDER
feature AIRMODE
feature ANTI_GRAVITY

# serial
serial 20 1 115200 57600 0 115200
serial 0 0 115200 57600 0 115200
serial 1 0 115200 57600 0 115200
serial 2 0 115200 57600 0 115200
serial 3 0 115200 57600 0 115200
serial 6 64 115200 57600 0 115200
serial 7 0 115200 57600 0 115200

# mixer
mixer QUADX

mmix reset


# servo
servo 0 1000 2000 1500 100 -1
servo 1 1000 2000 1500 100 -1
servo 2 1000 2000 1500 100 -1
servo 3 1000 2000 1500 100 -1
servo 4 1000 2000 1500 100 -1
servo 5 1000 2000 1500 100 -1
servo 6 1000 2000 1500 100 -1
servo 7 1000 2000 1500 100 -1

# servo mixer
smix reset


# beeper
beeper GYRO_CALIBRATED
beeper RX_LOST
beeper RX_LOST_LANDING
beeper DISARMING
beeper ARMING
beeper ARMING_GPS_FIX
beeper ARMING_GPS_NO_FIX
beeper BAT_CRIT_LOW
beeper BAT_LOW
beeper GPS_STATUS
beeper RX_SET
beeper ACC_CALIBRATION
beeper ACC_CALIBRATION_FAIL
beeper READY_BEEP
beeper MULTI_BEEPS
beeper DISARM_REPEAT
beeper ARMED
beeper SYSTEM_INIT
beeper ON_USB
beeper BLACKBOX_ERASE
beeper CRASH_FLIP
beeper CAM_CONNECTION_OPEN
beeper CAM_CONNECTION_CLOSE
beeper RC_SMOOTHING_INIT_FAIL

# beacon
beacon -RX_LOST
beacon -RX_SET

# map
map AETR1234

# led
led 0 0,0::C:0
led 1 0,0::C:0
led 2 0,0::C:0
led 3 0,0::C:0
led 4 0,0::C:0
led 5 0,0::C:0
led 6 0,0::C:0
led 7 0,0::C:0
led 8 0,0::C:0
led 9 0,0::C:0
led 10 0,0::C:0
led 11 0,0::C:0
led 12 0,0::C:0
led 13 0,0::C:0
led 14 0,0::C:0
led 15 0,0::C:0
led 16 0,0::C:0
led 17 0,0::C:0
led 18 0,0::C:0
led 19 0,0::C:0
led 20 0,0::C:0
led 21 0,0::C:0
led 22 0,0::C:0
led 23 0,0::C:0
led 24 0,0::C:0
led 25 0,0::C:0
led 26 0,0::C:0
led 27 0,0::C:0
led 28 0,0::C:0
led 29 0,0::C:0
led 30 0,0::C:0
led 31 0,0::C:0

# color
color 0 0,0,0
color 1 0,255,255
color 2 0,0,255
color 3 30,0,255
color 4 60,0,255
color 5 90,0,255
color 6 120,0,255
color 7 150,0,255
color 8 180,0,255
color 9 210,0,255
color 10 240,0,255
color 11 270,0,255
color 12 300,0,255
color 13 330,0,255
color 14 0,0,0
color 15 0,0,0

# mode_color
mode_color 0 0 1
mode_color 0 1 11
mode_color 0 2 2
mode_color 0 3 13
mode_color 0 4 10
mode_color 0 5 3
mode_color 1 0 5
mode_color 1 1 11
mode_color 1 2 3
mode_color 1 3 13
mode_color 1 4 10
mode_color 1 5 3
mode_color 2 0 10
mode_color 2 1 11
mode_color 2 2 4
mode_color 2 3 13
mode_color 2 4 10
mode_color 2 5 3
mode_color 3 0 8
mode_color 3 1 11
mode_color 3 2 4
mode_color 3 3 13
mode_color 3 4 10
mode_color 3 5 3
mode_color 4 0 7
mode_color 4 1 11
mode_color 4 2 3
mode_color 4 3 13
mode_color 4 4 10
mode_color 4 5 3
mode_color 5 0 0
mode_color 5 1 0
mode_color 5 2 0
mode_color 5 3 0
mode_color 5 4 0
mode_color 5 5 0
mode_color 6 0 6
mode_color 6 1 10
mode_color 6 2 1
mode_color 6 3 0
mode_color 6 4 0
mode_color 6 5 2
mode_color 6 6 3
mode_color 6 7 6
mode_color 6 8 0
mode_color 6 9 0
mode_color 6 10 0
mode_color 7 0 3

# aux
aux 0 0 0 900 900 0 0
aux 1 0 0 900 900 0 0
aux 2 0 0 900 900 0 0
aux 3 0 0 900 900 0 0
aux 4 0 0 900 900 0 0
aux 5 0 0 900 900 0 0
aux 6 0 0 900 900 0 0
aux 7 0 0 900 900 0 0
aux 8 0 0 900 900 0 0
aux 9 0 0 900 900 0 0
aux 10 0 0 900 900 0 0
aux 11 0 0 900 900 0 0
aux 12 0 0 900 900 0 0
aux 13 0 0 900 900 0 0
aux 14 0 0 900 900 0 0
aux 15 0 0 900 900 0 0
aux 16 0 0 900 900 0 0
aux 17 0 0 900 900 0 0
aux 18 0 0 900 900 0 0
aux 19 0 0 900 900 0 0

# adjrange
adjrange 0 0 0 900 900 0 0 0 0
adjrange 1 0 0 900 900 0 0 0 0
adjrange 2 0 0 900 900 0 0 0 0
adjrange 3 0 0 900 900 0 0 0 0
adjrange 4 0 0 900 900 0 0 0 0
adjrange 5 0 0 900 900 0 0 0 0
adjrange 6 0 0 900 900 0 0 0 0
adjrange 7 0 0 900 900 0 0 0 0
adjrange 8 0 0 900 900 0 0 0 0
adjrange 9 0 0 900 900 0 0 0 0
adjrange 10 0 0 900 900 0 0 0 0
adjrange 11 0 0 900 900 0 0 0 0
adjrange 12 0 0 900 900 0 0 0 0
adjrange 13 0 0 900 900 0 0 0 0
adjrange 14 0 0 900 900 0 0 0 0
adjrange 15 0 0 900 900 0 0 0 0
adjrange 16 0 0 900 900 0 0 0 0
adjrange 17 0 0 900 900 0 0 0 0
adjrange 18 0 0 900 900 0 0 0 0
adjrange 19 0 0 900 900 0 0 0 0
adjrange 20 0 0 900 900 0 0 0 0
adjrange 21 0 0 900 900 0 0 0 0
adjrange 22 0 0 900 900 0 0 0 0
adjrange 23 0 0 900 900 0 0 0 0
adjrange 24 0 0 900 900 0 0 0 0
adjrange 25 0 0 900 900 0 0 0 0
adjrange 26 0 0 900 900 0 0 0 0
adjrange 27 0 0 900 900 0 0 0 0
adjrange 28 0 0 900 900 0 0 0 0
adjrange 29 0 0 900 900 0 0 0 0

# rxrange
rxrange 0 1000 2000
rxrange 1 1000 2000
rxrange 2 1000 2000
rxrange 3 1000 2000

# vtxtable
vtxtable bands 0
vtxtable channels 0
vtxtable powerlevels 0
vtxtable powervalues
vtxtable powerlabels

# vtx
vtx 0 0 0 0 0 900 900
vtx 1 0 0 0 0 900 900
vtx 2 0 0 0 0 900 900
vtx 3 0 0 0 0 900 900
vtx 4 0 0 0 0 900 900
vtx 5 0 0 0 0 900 900
vtx 6 0 0 0 0 900 900
vtx 7 0 0 0 0 900 900
vtx 8 0 0 0 0 900 900
vtx 9 0 0 0 0 900 900

# rxfail
rxfail 0 a
rxfail 1 a
rxfail 2 a
rxfail 3 a
rxfail 4 h
rxfail 5 h
rxfail 6 h
rxfail 7 h
rxfail 8 h
rxfail 9 h
rxfail 10 h
rxfail 11 h
rxfail 12 h
rxfail 13 h
rxfail 14 h
rxfail 15 h
rxfail 16 h
rxfail 17 h

# master
set gyro_hardware_lpf = NORMAL
set gyro_lpf1_type = PT1
set gyro_lpf1_static_hz = 250
set gyro_lpf2_type = PT1
set gyro_lpf2_static_hz = 500
set gyro_notch1_hz = 0
set gyro_notch1_cutoff = 0
set gyro_notch2_hz = 0
set gyro_notch2_cutoff = 0
set gyro_calib_duration = 125
set gyro_calib_noise_limit = 48
set gyro_offset_yaw = 0
set gyro_overflow_detect = ALL
set yaw_spin_recovery = AUTO
set yaw_spin_threshold = 1950
set gyro_to_use = FIRST
set dyn_notch_count = 3
set dyn_notch_q = 300
set dyn_notch_min_hz = 100
set dyn_notch_max_hz = 600
set gyro_lpf1_dyn_min_hz = 250
set gyro_lpf1_dyn_max_hz = 500
set gyro_lpf1_dyn_expo = 5
set gyro_filter_debug_axis = ROLL
set acc_hardware = AUTO
set acc_lpf_hz = 25
set acc_trim_pitch = 0
set acc_trim_roll = 0
set acc_calibration = 0,0,0,0
set baro_bustype = I2C
set baro_spi_device = 0
set baro_i2c_device = 2
set baro_i2c_address = 119
set baro_hardware = AUTO
set mid_rc = 1500
set min_check = 1050
set max_check = 1900
set rssi_channel = 0
set rssi_src_frame_errors = OFF
set rssi_scale = 100
set rssi_offset = 0
set rssi_invert = OFF
set rssi_src_frame_lpf_period = 30
set rssi_smoothing = 125
set rc_smoothing = ON
set rc_smoothing_auto_factor = 30
set rc_smoothing_auto_factor_throttle = 30
set rc_smoothing_setpoint_cutoff = 0
set rc_smoothing_feedforward_cutoff = 0
set rc_smoothing_throttle_cutoff = 0
set rc_smoothing_debug_axis = ROLL
set fpv_mix_degrees = 0
set max_aux_channels = 14
set serialrx_provider = CRSF
set serialrx_inverted = OFF
set spektrum_sat_bind = 0
set spektrum_sat_bind_autoreset = ON
set srxl2_unit_id = 1
set srxl2_baud_fast = ON
set sbus_baud_fast = OFF
set crsf_use_negotiated_baud = OFF
set airmode_start_throttle_percent = 25
set rx_min_usec = 885
set rx_max_usec = 2115
set serialrx_halfduplex = OFF
set msp_override_channels_mask = 0
set adc_device = 1
set adc_vrefint_calibration = 0
set adc_tempsensor_calibration30 = 0
set adc_tempsensor_calibration110 = 0
set input_filtering_mode = OFF
set blackbox_sample_rate = 1/4
set blackbox_device = SPIFLASH
set blackbox_disable_pids = OFF
set blackbox_disable_rc = OFF
set blackbox_disable_setpoint = OFF
set blackbox_disable_bat = OFF
set blackbox_disable_alt = OFF
set blackbox_disable_rssi = OFF
set blackbox_disable_gyro = OFF
set blackbox_disable_gyrounfilt = OFF
set blackbox_disable_acc = OFF
set blackbox_disable_debug = OFF
set blackbox_disable_motors = OFF
set blackbox_disable_rpm = OFF
set blackbox_disable_gps = OFF
set blackbox_mode = NORMAL
set blackbox_high_resolution = OFF
set min_throttle = 1070
set max_throttle = 2000
set min_command = 1000
set motor_kv = 1960
set dshot_idle_value = 550
set dshot_burst = OFF
set dshot_bidir = OFF
set dshot_edt = OFF
set dshot_bitbang = AUTO
set dshot_bitbang_timer = AUTO
set use_unsynced_pwm = OFF
set motor_pwm_protocol = DSHOT600
set motor_pwm_rate = 480
set motor_pwm_inversion = OFF
set motor_poles = 14
set motor_output_reordering = 0,1,2,3,4,5,6,7
set thr_corr_value = 0
set thr_corr_angle = 800
set failsafe_delay = 15
set failsafe_off_delay = 10
set failsafe_throttle = 1000
set failsafe_switch_mode = STAGE1
set failsafe_throttle_low_delay = 100
set failsafe_procedure = DROP
set failsafe_recovery_delay = 5
set failsafe_stick_threshold = 30
set align_board_roll = 0
set align_board_pitch = 0
set align_board_yaw = 0
set gimbal_mode = NORMAL
set bat_capacity = 0
set vbat_max_cell_voltage = 430
set vbat_full_cell_voltage = 410
set vbat_min_cell_voltage = 330
set vbat_warning_cell_voltage = 350
set vbat_hysteresis = 1
set current_meter = ADC
set battery_meter = ADC
set vbat_detect_cell_voltage = 300
set use_vbat_alerts = ON
set use_cbat_alerts = OFF
set cbat_alert_percent = 10
set vbat_cutoff_percent = 100
set force_battery_cell_count = 0
set vbat_display_lpf_period = 30
set vbat_sag_lpf_period = 2
set ibat_lpf_period = 10
set vbat_duration_for_warning = 0
set vbat_duration_for_critical = 0
set vbat_scale = 110
set vbat_divider = 10
set vbat_multiplier = 1
set ibata_scale = 400
set ibata_offset = 0
set ibatv_scale = 0
set ibatv_offset = 0
set battery_continue = OFF
set beeper_inversion = ON
set beeper_od = OFF
set beeper_frequency = 0
set beeper_dshot_beacon_tone = 1
set yaw_motors_reversed = OFF
set mixer_type = LEGACY
set crashflip_motor_percent = 0
set crashflip_expo = 35
set rpm_limit = OFF
set rpm_limit_p = 25
set rpm_limit_i = 10
set rpm_limit_d = 8
set rpm_limit_value = 18000
set 3d_deadband_low = 1406
set 3d_deadband_high = 1514
set 3d_neutral = 1460
set 3d_deadband_throttle = 50
set 3d_limit_low = 1000
set 3d_limit_high = 2000
set 3d_switched_mode = OFF
set servo_center_pulse = 1500
set servo_pwm_rate = 50
set servo_lowpass_hz = 0
set tri_unarmed_servo = ON
set channel_forwarding_start = 4
set reboot_character = 82
set serial_update_rate_hz = 100
set imu_dcm_kp = 2500
set imu_dcm_ki = 0
set small_angle = 25
set imu_process_denom = 2
set auto_disarm_delay = 5
set gyro_cal_on_first_arm = OFF
set gps_provider = UBLOX
set gps_sbas_mode = NONE
set gps_auto_config = ON
set gps_auto_baud = OFF
set gps_ublox_acquire_model = STATIONARY
set gps_ublox_flight_model = AIRBORNE_4G
set gps_update_rate_hz = 10
set gps_ublox_utc_standard = AUTO
set gps_ublox_use_galileo = OFF
set gps_set_home_point_once = OFF
set gps_use_3d_speed = OFF
set gps_sbas_integrity = OFF
set gps_nmea_custom_commands = -
set gps_rescue_min_start_dist = 15
set gps_rescue_alt_mode = MAX_ALT
set gps_rescue_initial_climb = 10
set gps_rescue_ascend_rate = 750
set gps_rescue_return_alt = 30
set gps_rescue_ground_speed = 750
set gps_rescue_max_angle = 45
set gps_rescue_roll_mix = 150
set gps_rescue_pitch_cutoff = 75
set gps_rescue_imu_yaw_gain = 10
set gps_rescue_descent_dist = 20
set gps_rescue_descend_rate = 150
set gps_rescue_landing_alt = 4
set gps_rescue_disarm_threshold = 20
set gps_rescue_throttle_min = 1100
set gps_rescue_throttle_max = 1700
set gps_rescue_throttle_hover = 1275
set gps_rescue_sanity_checks = RESCUE_SANITY_FS_ONLY
set gps_rescue_min_sats = 8
set gps_rescue_allow_arming_without_fix = OFF
set gps_rescue_throttle_p = 15
set gps_rescue_throttle_i = 15
set gps_rescue_throttle_d = 20
set gps_rescue_velocity_p = 8
set gps_rescue_velocity_i = 40
set gps_rescue_velocity_d = 12
set gps_rescue_yaw_p = 20
set deadband = 0
set yaw_deadband = 0
set yaw_control_reversed = OFF
set pid_process_denom = 1
set runaway_takeoff_prevention = ON
set runaway_takeoff_deactivate_delay = 500
set runaway_takeoff_deactivate_throttle_percent = 20
set simplified_gyro_filter = ON
set simplified_gyro_filter_multiplier = 100
set tlm_inverted = OFF
set tlm_halfduplex = ON
set frsky_default_lat = 0
set frsky_default_long = 0
set frsky_gps_format = 0
set frsky_unit = METRIC
set frsky_vfas_precision = 0
set hott_alarm_int = 5
set pid_in_tlm = OFF
set report_cell_voltage = OFF
set ibus_sensor = 1,2,3,0,0,0,0,0,0,0,0,0,0,0,0
set mavlink_mah_as_heading_divisor = 0
set telemetry_disabled_voltage = OFF
set telemetry_disabled_current = OFF
set telemetry_disabled_fuel = OFF
set telemetry_disabled_mode = OFF
set telemetry_disabled_acc_x = OFF
set telemetry_disabled_acc_y = OFF
set telemetry_disabled_acc_z = OFF
set telemetry_disabled_pitch = OFF
set telemetry_disabled_roll = OFF
set telemetry_disabled_heading = OFF
set telemetry_disabled_altitude = OFF
set telemetry_disabled_vario = OFF
set telemetry_disabled_lat_long = OFF
set telemetry_disabled_ground_speed = OFF
set telemetry_disabled_distance = OFF
set telemetry_disabled_esc_current = ON
set telemetry_disabled_esc_voltage = ON
set telemetry_disabled_esc_rpm = ON
set telemetry_disabled_esc_temperature = ON
set telemetry_disabled_temperature = OFF
set telemetry_disabled_cap_used = ON
set ledstrip_visual_beeper = OFF
set ledstrip_visual_beeper_color = WHITE
set ledstrip_grb_rgb = GRB
set ledstrip_profile = STATUS
set ledstrip_race_color = ORANGE
set ledstrip_beacon_color = WHITE
set ledstrip_beacon_period_ms = 500
set ledstrip_beacon_percent = 50
set ledstrip_beacon_armed_only = OFF
set ledstrip_brightness = 100
set ledstrip_rainbow_delta = 0
set ledstrip_rainbow_freq = 120
set osd_units = METRIC
set osd_warn_bitmask = 8191
set osd_rssi_alarm = 20
set osd_link_quality_alarm = 80
set osd_rssi_dbm_alarm = -60
set osd_rsnr_alarm = 4
set osd_cap_alarm = 2200
set osd_alt_alarm = 100
set osd_distance_alarm = 0
set osd_esc_temp_alarm = 0
set osd_esc_rpm_alarm = -1
set osd_esc_current_alarm = -1
set osd_core_temp_alarm = 70
set osd_ah_max_pit = 20
set osd_ah_max_rol = 40
set osd_ah_invert = OFF
set osd_logo_on_arming = OFF
set osd_logo_on_arming_duration = 5
set osd_use_quick_menu = ON
set osd_show_spec_prearm = ON
set osd_tim1 = 2560
set osd_tim2 = 2561
set osd_vbat_pos = 234
set osd_rssi_pos = 234
set osd_link_quality_pos = 234
set osd_link_tx_power_pos = 234
set osd_rssi_dbm_pos = 234
set osd_rsnr_pos = 234
set osd_tim_1_pos = 234
set osd_tim_2_pos = 234
set osd_remaining_time_estimate_pos = 234
set osd_flymode_pos = 234
set osd_anti_gravity_pos = 234
set osd_g_force_pos = 234
set osd_throttle_pos = 234
set osd_vtx_channel_pos = 234
set osd_crosshairs_pos = 205
set osd_ah_sbar_pos = 206
set osd_ah_pos = 78
set osd_current_pos = 234
set osd_mah_drawn_pos = 234
set osd_wh_drawn_pos = 234
set osd_motor_diag_pos = 234
set osd_craft_name_pos = 234
set osd_pilot_name_pos = 234
set osd_gps_speed_pos = 234
set osd_gps_lon_pos = 234
set osd_gps_lat_pos = 234
set osd_gps_sats_pos = 234
set osd_home_dir_pos = 234
set osd_home_dist_pos = 234
set osd_flight_dist_pos = 234
set osd_compass_bar_pos = 234
set osd_altitude_pos = 234
set osd_pid_roll_pos = 234
set osd_pid_pitch_pos = 234
set osd_pid_yaw_pos = 234
set osd_debug_pos = 234
set osd_power_pos = 234
set osd_pidrate_profile_pos = 234
set osd_warnings_pos = 14665
set osd_avg_cell_voltage_pos = 234
set osd_pit_ang_pos = 234
set osd_rol_ang_pos = 234
set osd_battery_usage_pos = 234
set osd_disarmed_pos = 234
set osd_nheading_pos = 234
set osd_up_down_reference_pos = 205
set osd_ready_mode_pos = 234
set osd_nvario_pos = 234
set osd_esc_tmp_pos = 234
set osd_esc_rpm_pos = 234
set osd_esc_rpm_freq_pos = 234
set osd_rtc_date_time_pos = 234
set osd_adjustment_range_pos = 234
set osd_flip_arrow_pos = 234
set osd_core_temp_pos = 234
set osd_log_status_pos = 234
set osd_stick_overlay_left_pos = 234
set osd_stick_overlay_right_pos = 234
set osd_stick_overlay_radio_mode = 2
set osd_rate_profile_name_pos = 234
set osd_pid_profile_name_pos = 234
set osd_profile_name_pos = 234
set osd_rcchannels_pos = 234
set osd_camera_frame_pos = 35
set osd_efficiency_pos = 234
set osd_total_flights_pos = 234
set osd_aux_pos = 234
set osd_sys_goggle_voltage_pos = 234
set osd_sys_vtx_voltage_pos = 234
set osd_sys_bitrate_pos = 234
set osd_sys_delay_pos = 234
set osd_sys_distance_pos = 234
set osd_sys_lq_pos = 234
set osd_sys_goggle_dvr_pos = 234
set osd_sys_vtx_dvr_pos = 234
set osd_sys_warnings_pos = 234
set osd_sys_vtx_temp_pos = 234
set osd_sys_fan_speed_pos = 234
set osd_stat_bitmask = 1879062316
set osd_profile = 1
set osd_profile_1_name = -
set osd_profile_2_name = -
set osd_profile_3_name = -
set osd_gps_sats_show_hdop = OFF
set osd_displayport_device = AUTO
set osd_rcchannels = -1,-1,-1,-1
set osd_camera_frame_width = 24
set osd_camera_frame_height = 11
set osd_stat_avg_cell_value = OFF
set osd_framerate_hz = 12
set osd_menu_background = TRANSPARENT
set osd_aux_channel = 1
set osd_aux_scale = 200
set osd_aux_symbol = 65
set osd_canvas_width = 30
set osd_canvas_height = 13
set osd_craftname_msgs = OFF
set task_statistics = ON
set debug_mode = NONE
set rate_6pos_switch = OFF
set pwr_on_arm_grace = 5
set enable_stick_arming = OFF
set vtx_band = 0
set vtx_channel = 0
set vtx_power = 0
set vtx_low_power_disarm = OFF
set vtx_softserial_alt = OFF
set vtx_freq = 0
set vtx_pit_mode_freq = 0
set vtx_halfduplex = ON
set vtx_spi_bus = 0
set vcd_video_system = AUTO
set displayport_msp_col_adjust = 0
set displayport_msp_row_adjust = 0
set displayport_msp_fonts = 0,1,2,3
set displayport_msp_use_device_blink = OFF
set esc_sensor_halfduplex = OFF
set esc_sensor_current_offset = 0
set led_inversion = 0
set dashboard_i2c_bus = 0
set dashboard_i2c_addr = 60
set rangefinder_hardware = NONE
set pinio_config = 1,1,1,1
set pinio_box = 255,255,255,255
set usb_hid_cdc = OFF
set usb_msc_pin_pullup = ON
set flash_spi_bus = 2
set rcdevice_init_dev_attempts = 6
set rcdevice_init_dev_attempt_interval = 1000
set rcdevice_protocol_version = 0
set rcdevice_feature = 0
set gyro_1_bustype = SPI
set gyro_1_spibus = 1
set gyro_1_i2cBus = 0
set gyro_1_i2c_address = 0
set gyro_1_sensor_align = CW90
set gyro_1_align_roll = 0
set gyro_1_align_pitch = 0
set gyro_1_align_yaw = 900
set gyro_2_bustype = NONE
set gyro_2_spibus = 0
set gyro_2_i2cBus = 0
set gyro_2_i2c_address = 0
set gyro_2_sensor_align = DEFAULT
set gyro_2_align_roll = 0
set gyro_2_align_pitch = 0
set gyro_2_align_yaw = 0
set i2c1_pullup = ON
set i2c1_clockspeed_khz = 800
set i2c2_pullup = ON
set i2c2_clockspeed_khz = 800
set i2c3_pullup = ON
set i2c3_clockspeed_khz = 800
set i2c4_pullup = ON
set i2c4_clockspeed_khz = 800
set scheduler_relax_rx = 25
set scheduler_relax_osd = 25
set serialmsp_halfduplex = OFF
set timezone_offset_minutes = 0
set rpm_filter_harmonics = 3
set rpm_filter_weights = 100,100,100
set rpm_filter_q = 500
set rpm_filter_min_hz = 100
set rpm_filter_fade_range_hz = 50
set rpm_filter_lpf_hz = 150
set stats_min_armed_time_s = -1
set stats_total_flights = 0
set stats_total_time_s = 0
set stats_total_dist_m = 0
set stats_mah_used = 0
set craft_name = -
set pilot_name = -
set altitude_source = DEFAULT
set altitude_prefer_baro = 100
set altitude_lpf = 300
set altitude_d_lpf = 100
set box_user_1_name = -
set box_user_2_name = -
set box_user_3_name = -
set box_user_4_name = -

profile 0

# profile 0
set profile_name = -
set dterm_lpf1_dyn_min_hz = 75
set dterm_lpf1_dyn_max_hz = 150
set dterm_lpf1_dyn_expo = 5
set dterm_lpf1_type = PT1
set dterm_lpf1_static_hz = 75
set dterm_lpf2_type = PT1
set dterm_lpf2_static_hz = 150
set dterm_notch_hz = 0
set dterm_notch_cutoff = 0
set vbat_sag_compensation = 0
set pid_at_min_throttle = ON
set anti_gravity_gain = 80
set anti_gravity_cutoff_hz = 5
set anti_gravity_p_gain = 100
set acc_limit_yaw = 0
set acc_limit = 0
set crash_dthreshold = 50
set crash_gthreshold = 400
set crash_setpoint_threshold = 350
set crash_time = 500
set crash_delay = 0
set crash_recovery_angle = 10
set crash_recovery_rate = 100
set crash_limit_yaw = 200
set crash_recovery = OFF
set iterm_rotation = OFF
set iterm_relax = RP
set iterm_relax_type = SETPOINT
set iterm_relax_cutoff = 15
set iterm_windup = 85
set iterm_limit = 400
set pidsum_limit = 500
set pidsum_limit_yaw = 400
set yaw_lowpass_hz = 100
set throttle_boost = 5
set throttle_boost_cutoff = 15
set p_pitch = 47
set i_pitch = 84
set d_pitch = 46
set f_pitch = 125
set p_roll = 45
set i_roll = 80
set d_roll = 40
set f_roll = 120
set p_yaw = 45
set i_yaw = 80
set d_yaw = 0
set f_yaw = 120
set angle_p_gain = 50
set angle_feedforward = 50
set angle_feedforward_smoothing_ms = 80
set angle_limit = 60
set angle_earth_ref = 100
set horizon_level_strength = 75
set horizon_limit_sticks = 75
set horizon_limit_degrees = 135
set horizon_ignore_sticks = OFF
set horizon_delay_ms = 500
set abs_control_gain = 0
set abs_control_limit = 90
set abs_control_error_limit = 20
set abs_control_cutoff = 11
set use_integrated_yaw = OFF
set integrated_yaw_relax = 200
set d_min_roll = 30
set d_min_pitch = 34
set d_min_yaw = 0
set d_max_gain = 37
set d_max_advance = 20
set motor_output_limit = 100
set auto_profile_cell_count = 0
set launch_control_mode = NORMAL
set launch_trigger_allow_reset = ON
set launch_trigger_throttle_percent = 20
set launch_angle_limit = 0
set launch_control_gain = 40
set thrust_linear = 0
set transient_throttle_limit = 0
set feedforward_transition = 0
set feedforward_averaging = OFF
set feedforward_smooth_factor = 25
set feedforward_jitter_factor = 7
set feedforward_boost = 15
set feedforward_max_rate_limit = 90
set dyn_idle_min_rpm = 0
set dyn_idle_p_gain = 50
set dyn_idle_i_gain = 50
set dyn_idle_d_gain = 50
set dyn_idle_max_increase = 150
set dyn_idle_start_increase = 50
set level_race_mode = OFF
set simplified_pids_mode = RPY
set simplified_master_multiplier = 100
set simplified_i_gain = 100
set simplified_d_gain = 100
set simplified_pi_gain = 100
set simplified_dmax_gain = 100
set simplified_feedforward_gain = 100
set simplified_pitch_d_gain = 100
set simplified_pitch_pi_gain = 100
set simplified_dterm_filter = ON
set simplified_dterm_filter_multiplier = 100
set tpa_mode = D
set tpa_rate = 65
set tpa_breakpoint = 1350
set tpa_low_rate = 20
set tpa_low_breakpoint = 1050
set tpa_low_always = OFF
set ez_landing_threshold = 25
set ez_landing_limit = 5

rateprofile 0

# rateprofile 0
set rateprofile_name = -
set thr_mid = 50
set thr_expo = 0
set rates_type = ACTUAL
set quickrates_rc_expo = OFF
set roll_rc_rate = 7
set pitch_rc_rate = 7
set yaw_rc_rate = 7
set roll_expo = 0
set pitch_expo = 0
set yaw_expo = 0
set roll_srate = 67
set pitch_srate = 67
set yaw_srate = 67
set throttle_limit_type = OFF
set throttle_limit_percent = 100
set roll_rate_limit = 1998
set pitch_rate_limit = 1998
set yaw_rate_limit = 1998

# end the command batch
batch end
```